### PR TITLE
Add a newLength parameter to transferToImmutable

### LINKED ISSUE #15: Should `transferToImmutable` support a _newByteLength_ argument?
In favor: Least surprise. Minimize consequences of reasonable surprise. If we add this parameter, those who expect it not to be there and don't provide such an argument get what they expect anyway. OTOH, if we omit it, those who expect it to be there and provide such an argument do not get either a static or dynamic diagnostic. Rather, the code proceeds to silently do something other than they expect.

Against: It might be useful for truncation, but probably rarely for expansion because the output would have an all-zeroes suffix that could not be changed.

Both sides can also make arguments from orthogonality: 
- In favor is more orthogonal because the parameter exists and means the same thing across all `transfer*` operations. 
- Against is more orthogonal because making a buffer immutable is distinct from changing its size, and can be painlessly composed with a transfer operation which does change the size, while still being zero-copy.

### PR DESCRIPTION
**Pull request by gibson042**
As preferred in the plenary discussion.

Fixes #15

### CODE REVIEW

File: spec.emu [hunk 1]
```diff
@@ -637,11 +637,11 @@ contributors: Mark S. Miller, Richard Gibson
       <ins class="block">
 
         <emu-clause id="sec-arraybuffer.prototype.transfertoimmutable">
-          <h1>ArrayBuffer.prototype.transferToImmutable ( )</h1>
+          <h1>ArrayBuffer.prototype.transferToImmutable ( [ _newLength_ ] )</h1>
           <p>This method performs the following steps when called:</p>
           <emu-alg>
             1. Let _O_ be the *this* value.
-            1. Return ? ArrayBufferCopyAndDetach(_O_, *undefined*, ~immutable~).
+            1. Return ? ArrayBufferCopyAndDetach(_O_, _newLength_, ~immutable~).
```
Comment (ljharb) [inline spec.emu:644]: this causes it to be coerced (using ToIndex), which violates our new principles, so it should be validated to be an index first i think? eg:
```suggestion
            1. Let _newIndex_ be ? ToIndex(_newLength_);
            1. If _newIndex_ is not _newLength_, throw a TypeError exception.
            1. Return ? ArrayBufferCopyAndDetach(_O_, _newIndex_, ~immutable~).
```

Comment (gibson042) [reply spec.emu:644]: A big argument for including the parameter at all was consistency with existing `transfer` and `transferToFixedLength` methods, so I really don't think its treatment should differ.

Comment (bakkot) [reply spec.emu:644]: For things which are close cousins of existing APIs, which this is, we said we were generally going to stick with the behavior from the original API.

Comment (ljharb) [reply spec.emu:644]: Fair enough.

**Review by erights — APPROVED** [review]
LGTM, thanks!

**Pull request merged**